x-pack/plugins/ml/public/application/trained_models/models_management/test_models/models/inference_base.ts

```diff
@@ -60,19 +61,48 @@ export abstract class InferenceBase<TInferResponse> {
   protected abstract readonly inferenceTypeLabel: string;
   protected inputField: string;
   protected readonly modelInputField: string;
-  private inputText$ = new BehaviorSubject<string[]>([]);
+  protected inputText$ = new BehaviorSubject<string[]>([]);
   private inferenceResult$ = new BehaviorSubject<TInferResponse[] | null>(null);
   private inferenceError$ = new BehaviorSubject<MLHttpFetchError | null>(null);
   private runningState$ = new BehaviorSubject<RUNNING_STATE>(RUNNING_STATE.STOPPED);
+  protected inputTextValid$ = new BehaviorSubject<boolean>(false);
+  protected isValid$ = new BehaviorSubject<boolean>(false);
   protected readonly info: string[] = [];
 
+  private validators$: Array<Observable<boolean>> = [];
+  private validatorsSubscriptions$: Subscription = new Subscription();
+
   constructor(
     protected readonly trainedModelsApi: ReturnType<typeof trainedModelsApiProvider>,
     protected readonly model: estypes.MlTrainedModelConfig,
     protected readonly inputType: INPUT_TYPE
   ) {
     this.modelInputField = model.input?.field_names[0] ?? DEFAULT_INPUT_FIELD;
     this.inputField = this.modelInputField;
+
+    this.validators$.push(
+      this.inputText$.pipe(map((inputText) => inputText.some((t) => t !== '')))
+    );
+  }
+
+  public destroy() {
+    this.validatorsSubscriptions$.unsubscribe();
+  }
+
+  protected initializeValidators(additionalValidators?: Array<Observable<boolean>>) {
```

Comment:
Is there a reason why we don't check the running state before enabling the `Test` button?

Reply:
The test flyout should not be available when a ob isn't started. but there appears to be a bug with this check.

Reply:
Fixed in https://github.com/elastic/kibana/pull/145357/commits/bbf1baa2dcc8ba0f3acc13415520b0dd04387076